scipy/optimize/_linprog_rs.py

```diff
@@ -333,7 +333,7 @@ def _phase_two(c, A, x, b, callback, postsolve_args, maxiter, tol, disp,
     else:
         B = LU(A, b)
 
-    for iteration in range(iteration, iteration + maxiter):
```

Comment:
`for` loops are better to have a definite ending in case something weird happens to `maxiter` variable that causes continous running and moreover you don't need to keep the iteration counts that are essentially C habits. But up-to-you this is just a nitpick.

Reply:
I agree. I wrote the original code and I changed it here to fix a bug in the iteration count. Right now, I can't for the life of me see why I couldn't do the same thing with the `for` loop. But I'm hesitant to change it right now if this is optional.

Reply:
I couldn't leave it alone. Fixed in 7f5e830b59cb2371c088872c5c6a197d4eb99f17.